### ISSUE TITLE
Future Features, Milestones and source code

### ISSUE THREAD

**Issue by 3XistencE** [libretro/dosbox-svn#43]
Hi,
i remember some milestones about DosBox on RetroArch and, in case somebody doesn't know, point out that there is actually as binary, but also as source code, a DosBox SVN version compiled already with those additional features:
+Emulation of a 3Dfx voodoo card with internal (software or OpenGL) or external wrapper
+improved emulation of OPL3 (a chip for FM sound synthesis from Yamaha)
+more balanced sound and more authentic volume of FM sound reproduction
+improved sound of PC speaker emulation
+Roland MT-32 midi synthesizer emulation
+Integration of Fluidsynth (a software midi synthesizer with Soundfont support)
+Support for up to 10 joystick axes and 2 D-Pads (2 complete 360-compatible controllers)+Mouse keys can also be placed on keyboard keys or controller buttons in the mapper
+Support of up to 384 MB main memory, necessary for the use of Windows 9x with DOSBo
+4MB video memory, reduces sprite flickering in games with the build engine (“Duke Nukem 3D”, “Blood”, etc.)
+Support of MSF time stamps in CUE files and support of MP3, FLAC, WAV and OPUS files as compressed audio files

I think it can be usefull and time saving to check the source in order to have a better idea about how to add those features (if i remember well the 3dfx was a future milestone to reach for DosBox on RetroArch or ask directly the guy if he want to collaborate.

[link to downloadable source code](https://dosboxdl.yesterplay.net/?do=download&file=DOSBox%20ECE%20r4334%20(source).7z)

**Comment by realnc** [comment]
I added much of this to this core:

https://github.com/realnc/dosbox-core

It's not in the RA updater though. You need to compile it from source. I was told the RA build bot is overloaded and can't take more cores so it can't be in the updater.

**Comment by 3XistencE** [comment]
A Wonderful job @realnc !
So it's already with all those stuff but simply is not merged with the main repository?
I'm really curious to test it and feel myself back with a 486dx2 (i will ever miss the "turbo" button on it).

What are the next milestones/features to add?

**Comment by realnc** [comment]
It will never be merged. It's a new core that's independent of this one. As to when it will be added to RA, I don't know. I'll try and ask around on the libretro discord about the buildbot.

If you want to try, the latest build is here: http://83.212.109.87/~realnc/dosbox-core/

**Comment by twinaphex** [comment]
I'm OK with a new core being added. I think I communicated this the last time around too.

Let's decide on a name for the new core. dosbox-core? Can you also explain how it will differentiate itself from Dosbox SVN so that we know what to tell users?

**Comment by 3XistencE** [comment]
Well the quantity of feature added is huge: mainly dosbox SVN is a fork from the main dosboxsvn and this version is a modded/patched one with additional features not already added on dosboxsvn.

@realnc think at a nice name for your modified core =)
Does this version will be also 32bit?

Also if it's possible i suggest to add the feature, especially because of the nature of RetroArch itself, inside the dosboxsvn core options about the quantity of -freesize (1024 or some selectable default value between 1mb to also 800mb for CD-rom installations) at the end of pre-done mount function.  This will let the final user to install or save without problem using dosbox as it is.
Can be good also to add a selectable mount directory (as sometimes our old dos stuff needs to be into a specific directory)

Right about now there's no free memory and some software go crazy without free memory.

**Comment by realnc** [comment]
@twinaphex 

> I'm OK with a new core being added. I think I communicated this the last time around too.

You mentioned that the buildbot was not ready yet. So it's ready now?

> Let's decide on a name for the new core. dosbox-core?

Well we can keep the name, unless you think it's too confusing when it appears in the updater and users don't know whether to use "-SVN" or "-core".

> Can you also explain how it will differentiate itself from Dosbox SVN

There's added features (native midi on windows/linux, as well as fluidsynth, munt mt-32, bassmidi and cycle-accurate opl3) as well as some improvements and bugfixes. Future additions and improvements will go into the new core.

**Comment by realnc** [comment]
@3XistencE Yes, the core works fine on 32-bit as well. I now uploaded a 32-bit windows build as well to the above URL.

As for suggestions or bugreports, you should open new ones here: https://github.com/realnc/dosbox-core/issues
